eventhubs/receive_eph.go

```diff
@@ -0,0 +1,114 @@
+	"context"
+	"fmt"
+	"log"
+	"sync"
+	"time"
+
+	aad "github.com/Azure/azure-amqp-common-go/aad"
+	eventhubs "github.com/Azure/azure-event-hubs-go"
+	eph "github.com/Azure/azure-event-hubs-go/eph"
+	storageLeaser "github.com/Azure/azure-event-hubs-go/storage"
+	azure "github.com/Azure/go-autorest/autorest/azure"
+
+	// imports within this repo
+	"github.com/Azure-Samples/azure-sdk-for-go-samples/helpers"
+	storage "github.com/Azure-Samples/azure-sdk-for-go-samples/storage"
+)
+
+func ReceiveViaEPH() {
+	ctx := context.Background()
+
+	// get a struct with metadata about the Azure environment
+	azureEnv, err := azure.EnvironmentFromName("AzurePublicCloud")
+	if err != nil {
+		log.Fatalf("could not get azure.Environment struct: %s\n", err)
+	}
+
+	// create an access token provider using AAD principal defined in environment
+	tokenProvider, err := aad.NewJWTProvider(aad.JWTProviderWithEnvironmentVars())
+	if err != nil {
+		log.Fatalf("failed to configure AAD JWT provider: %s\n", err)
+	}
+
+	// create a storage account and container to maintain dictionary of leases
+	// and checkpoints
+	log.Printf("creating storage account\n")
+	_, err = storage.CreateStorageAccount(ctx, storageAccountName)
+	if err != nil {
+		log.Fatalf("could not create storage account: %s\n", err)
+	}
+	log.Printf("creating storage container\n")
+	_, err = storage.CreateContainer(ctx, storageAccountName, storageContainerName)
+	if err != nil {
+		log.Fatalf("could not create storage container: %s\n", err)
+	}
+
+	// use helper method to exchange AAD credentials for SAS token
+	cred, err := storageLeaser.NewAADSASCredential(
+		helpers.SubscriptionID(),
+		helpers.ResourceGroupName(),
+		storageAccountName,
+		storageContainerName,
+		storageLeaser.AADSASCredentialWithEnvironmentVars())
+	if err != nil {
+		log.Fatalf("could not prepare a storage credential: %s\n", err)
+	}
+
+	// create a leaser and checkpointer backed by a storage container
+	leaserCheckpointer, err := storageLeaser.NewStorageLeaserCheckpointer(
+		cred,
+		storageAccountName,
+		storageContainerName,
+		azureEnv)
+	if err != nil {
+		log.Fatalf("could not prepare a storage leaserCheckpointer: %s\n", err)
+	}
+
+	// use all of the above to create an Event Processor
+	p, err := eph.New(
+		ctx,
+		nsName,
+		hubName,
+		tokenProvider,
+		leaserCheckpointer,
+		leaserCheckpointer,
+		eph.WithNoBanner())
+	if err != nil {
+		log.Fatalf("failed to create EPH: %s\n", err)
+	}
+	defer p.Close(context.Background())
+
+	// set up a handler for the Event Processor which will receive a single
+	// message, print it to the console, and allow the process to exit
+
+	// set up wait group to wait for expected message
+	var wg sync.WaitGroup
+	wg.Add(1) // wait for one message
+
+	handler := func(ctx context.Context, event *eventhubs.Event) error {
+		fmt.Printf("received: %s\n", string(event.Data))
+		wg.Done()
+		return nil
+	}
+
+	// register the handler with the Event Processor
+	close, err := p.Receive(handler)
```

Comment:
@joshgav I liked your suggestion of changing this to register and unregister handler. I'm going to make that change to better illustrate the difference between Receive and Start*

Reply:
:+1: Great! Should I wait on that change to merge? If it will be a little while we can update this when its ready.